tests/acceptance/01_vars/02_functions/execresult_select.cf

```diff
@@ -0,0 +1,62 @@
+#######################################################
+#
+# Test that you can select stderr/stdout in execresult
+#
+#######################################################
+
+body common control
+{
+      inputs => { "../../default.cf.sub" };
+      bundlesequence  => { default("$(this.promise_filename)") };
+      version => "1.0";
+}
+
+#######################################################
+
+bundle agent test
+{
+  meta:
+      "description" -> { "CFE-3108" }
+        string => "Test that you can select stderr/stdout in execresult";
+
+  vars:
+      "result_with_stdout_stderr"
+        string => execresult("$(G.echo) stdout; $(G.echo) stderr >&2", "useshell", "both");
+
+      "result_with_stdout"
+        string => execresult("$(G.echo) stdout; $(G.echo) stderr >&2", "useshell", "stdout");
+
+      "result_with_stderr"
+        string => execresult("$(G.echo) stdout; $(G.echo) stderr >&2", "useshell", "stderr");
+
+  classes:
+      "pass_stdout_stderr"
+        expression => strcmp( "stdout$(const.n)stderr", $(result_with_stdout_stderr) );
+
+      "pass_stdout"
+        expression => strcmp( "stdout", $(result_with_stdout) );
+
+      "pass_stderr"
+        expression => strcmp( "stderr", $(result_with_stderr) );
+
+  reports:
+
+    !pass_stdout_stderr::
+      "result_with_stdout_stderr == '$(result_with_stdout_stderr)', expected 'stdout$(const.n)stderr'";
+    !pass_stdout::
+      "result_with_stdout == '$(result_with_stdout_stderr)', expected 'stdout'";
+    !pass_stderr::
+      "result_with_stderr == '$(result_with_stderr)', expected 'stderr'";
+
+    pass_stdout_stderr.pass_stdout.pass_stderr::
+      "$(this.promise_filename) Pass";
```

Comment:
So that soft failing can be supported (not that it's necessary here, but for the next copy pasta) I prefer to use the DCS bundles to pass/fail.

    methods:
    
     "Pass"
       usebundle => dcs_pass($(this.promise_filename)),
       if => "pass_stdout_stderr.pass_stdout.pass_stderr";
    
     "Fail"
       usebundle => dcs_fail($(this.promise_filename)),
       if => "!(pass_stdout_stderr.pass_stdout.pass_stderr)";
    
     # Or, even better I think:
     "Pass/Fail" usebundle => dcs_passif( "pass_stdout_stderr.pass_stdout.pass_stderr", $(this.promise_filename) );

Reply:
Noted.

Reply:
@vpodzime @nickanderson https://github.com/cfengine/core/pull/4266